.github/workflows/CI.yml

```diff
@@ -196,8 +196,15 @@ jobs:
     runs-on: ubuntu-22.04
     permissions:
       contents: write
+      id-token: write  # IMPORTANT: mandatory for trusted publishing
+
+    environment:  # Environment for trusted publishing
+      name: test.pypi
+      url: https://test.pypi.org/p/gcovr
```

Comment:
These privileges should only be given to a separate publishing job which should only contain downloading the dists and publishing them. The transitive build deps and any other software shouldn't have access to OIDC due to future risks of impersonation / privilege escalation. Additionally, things like checking if the dists are buildable should be running continuously, on PRs, in contexts where releasing isn't happening.